src/test/java/org/jboss/remoting3/test/MaxChannelsTestCase.java

```diff
@@ -0,0 +1,207 @@
+        domainBuilder.setPermissionMapper((permissionMappable, roles) -> PermissionVerifier.ALL);
+        builder.setSecurityDomain(domainBuilder.build());
+        builder.setFactory(saslServerFactory);
+        builder.setMechanismConfigurationSelector(mechanismInformation -> SaslMechanismInformation.Names.SCRAM_SHA_256.equals(mechanismInformation.getMechanismName()) ? MechanismConfiguration.EMPTY : null);
+        final SaslAuthenticationFactory saslAuthenticationFactory = builder.build();
+        // The default OptionMap defined in EndPoint should be honored even the OptionMap in createServer method does not specify any.
+        server = networkServerProvider.createServer(new InetSocketAddress("localhost", 30123), OptionMap.create(Options.SSL_ENABLED, Boolean.FALSE), saslAuthenticationFactory, SSLContext.getDefault());
+    }
+
+    @After
+    public void after() {
+        IoUtils.safeClose(server);
+        IoUtils.safeClose(clientEndpoint);
+        IoUtils.safeClose(serverEndpoint);
+        System.gc();
+        System.runFinalization();
+        Logger.getLogger("TEST").infof("Finished test %s", name.getMethodName());
+    }
+
+    private int getMaxInboundChannelsCount() {
+        return 50;
+    }
+
+    private OptionMap optionMap() {
+        return OptionMap.builder()
+                .set(RemotingOptions.MAX_INBOUND_CHANNELS, getMaxInboundChannelsCount())
+                .set(RemotingOptions.MAX_OUTBOUND_CHANNELS, getMaxInboundChannelsCount())
+                .set(Options.SSL_ENABLED, Boolean.FALSE)
+                .getMap();
+    }
+
+    @Test
+    public void testMaxInboundChannels() throws Exception {
+        final XnioWorker clientWorker = clientEndpoint.getXnioWorker();
+        final Queue<Throwable> problems = new ConcurrentLinkedQueue<Throwable>();
+        final AtomicReferenceArray<Channel> clientChannels = new AtomicReferenceArray<>(getMaxInboundChannelsCount());
+        final CountDownLatch clientChannelCount = new CountDownLatch(getMaxInboundChannelsCount() + 1);
+        final AtomicInteger openedChannels = new AtomicInteger(0);
+        serverEndpoint.registerService("test", new OpenListener() {
+            public void channelOpened(final Channel channel) {
+                Logger.getLogger("TEST").infof("Channel counts: %s", openedChannels.incrementAndGet());
+            }
+            public void registrationTerminated() {
+            }
+        }, OptionMap.EMPTY);
+        IoFuture<Connection> futureConnection = AuthenticationContext.empty().with(MatchRule.ALL, AuthenticationConfiguration.empty().useName("bob").usePassword("pass").setSaslMechanismSelector(SaslMechanismSelector.NONE.addMechanism("SCRAM-SHA-256"))).run(new PrivilegedAction<IoFuture<Connection>>() {
+            public IoFuture<Connection> run() {
+                try {
+                    return clientEndpoint.connect(new URI("remote://localhost:30123"), optionMap());
```

Comment:
To really test that the defaultConnectionsOptionMap value is getting picked up by outgoing connection creation, shouldn't the OptionMap passed be be OptionMap.EMPTY?

Reply:
Done in https://github.com/tadamski/jboss-remoting/pull/2